packages/mobile/src/components/overflow-menu-drawer/TrackOverflowMenuDrawer.tsx

```diff
@@ -81,6 +86,18 @@ const TrackOverflowMenuDrawer = ({ render }: Props) => {
     getUser(state, { id: track?.owner_id })
   )
 
+  const handlePurchasePress = useCallback(() => {
+    if (track?.track_id) {
+      openPremiumContentPurchaseModal(
+        {
+          contentId: track?.track_id,
+          contentType: PurchaseableContentType.TRACK
+        },
+        { source: ModalSource.LineUpCollectionTile }
```

Comment:
I think we should make this it's own source.
Also, `Collection` would be incorrect to use

Reply:
hm i see `TrackListItem` could that work?